pkg/arvo/sys/vane/clay.hoon

```diff
@@ -149,21 +149,84 @@
   $%  [%0 =desk =lobe]
       [%1 =desk =lobe]
   ==
+::
 +$  fell
   $%  [%direct p=lobe q=page]
       [%delta p=lobe q=[p=mark q=lobe] r=page]
       [%dead p=lobe ~]
       [%1 peg=(unit page)]
   ==
 ::
-::  Ford cache
+::  Global ford cache
+::
+::    Refcount includes references from other items in the cache, and
+::    from spills in each desk
+::
+::    This is optimized for minimizing the number of rebuilds, and given
+::    that, minimizing the amount of memory used.  It is relatively slow
+::    to lookup, because generating a cache key can be fairly slow (for
+::    files, it requires parsing; for tubes, it even requires building
+::    the marks).
+::
++$  flow  (map leak [refs=@ud =soak])
+::
+::  Per-desk ford cache
+::
+::    Spill is the set of "roots" we have into the global ford cache.
+::    We add a root for everything referenced directly or indirectly on
+::    a desk, then invalidate them on commit only if their dependencies
+::    change.
+::
+::    Sprig is a fast-lookup index over the global ford cache.  The only
+::    goal is to make cache hits fast.
+::
++$  flue  [spill=(set leak) sprig=(map mist [=leak =soak])]
+::
+::  Ford build without content.
+::
++$  mist
+  $%  [%file =path]
+      [%nave =mark]
+      [%dais =mark]
+      [%cast =mars]
+      [%tube =mars]
+      [%vale =path]
+      [%arch =path]
+  ==
+::
+::  Ford build with content.
+::
++$  pour
+  $%  [%file =path]
+      [%nave =mark]
+      [%dais =mark]
+      [%cast =mars]
+      [%tube =mars]
+      ::  leafs
+      ::
+      [%vale =path lob=(unit lobe)]
```

Comment:
why is `lob` a unit?  I would have expected to always have a value for the file contents.

Reply:
there was a version where it could depend on a file not existing, but that's gone now, so i'll change it